### DISCUSSION TITLE
[Kavita] New Feature - Native WebDAV support

### DISCUSSION THREAD

**Discussion by capelo0** [Kareadita/Kavita#3304] [Ideas]
### Idea Description

My entire eBook library is stored in Nextcloud, and currently, there is no way to visualize or manage these eBooks through Kavita without complex and unreliable workarounds. I've tried rclone to mount WebDAV, but the contents are not passing to Kavita's docker container.
As Nextcloud provides WebDAV access, I'm kindly requesting to get in Kavitha native WebDAV support. 
This would be a game changer capability allowing users to connect their WebDAV-enabled storage solutions directly to the application.

### Idea Category

Feature Enhancement

### Duration of Using Kavita

I can't use yet because it does not support my need.

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by capelo0** [comment]
Why was it tagged with "not-likely"?

**Reply by majora2007** [reply]
From what I read about it, it looks like I'll have to add specific code to support WebDAV. I'm not interested in maintaining custom code for this. Hence the tag. Kavita supports the filesystem as its seamless to the software.

**Reply by capelo0** [reply]
Thank you for the quick reply @majora2007 and for investigating the subject. 
I understand the reason and I'll try to bring this request to other similar platforms, since at this moment there is no solution to address it and the growth of users leveraging nextcloud or other personal cloud solutions is huge, requiring this kind of support.
Once again, thank you for your time.

**Reply by pprkut** [reply]
On linux you could probably use something like https://github.com/volga629/davfs2 or https://github.com/miquels/webdavfs to mount the WebDAV share locally. Windows does seem to have [something similar](https://help.dreamhost.com/hc/en-us/articles/216473357-Accessing-WebDAV-with-Windows).
Kavita could then access it as though it was on the local filesystem.

**Reply by DieselTech** [reply]
To add to the options pprkut gave, rclone runs on basically every platform and has support for webdav

https://rclone.org/webdav/

This would let you connect to a webdav service and expose it to the local machine as a filesystem.

**Reply by pprkut** [reply]
Well, @capelo0 did mention that rclone wasn't working for them. Not knowing the exact error I can't speak to whether my suggestions would have worked either, but maybe using `--allow-other` with rclone is all it would take?

**Reply by DieselTech** [reply]
Sure but that is a problem with the users configuration, it isn't rclone not being able to support what they are doing. If there is rclone issues they should reach out to whatever support team they have.

**Reply by therobbiedavis** [reply]
A Kavita docker container reading a WebDAV mount via rclone is exactly how I have mine setup on ubuntu. If WebDAV on rclone isn't working then it's likely a custom implementation of WebDAV would be unlikely to work either. The real solution isn't a custom implementation but rather fixing what's wrong with rclone.

**Reply by sud0xenn** [reply]
Are there any existing native solutions which support webdav streaming either directly in browser or via their app? I really don't want to mount a hosted server as a network drive to be able to read from it ^^"
